### PR TITLE
ZOOKEEPER-3145. Refactor QuorumPeerMainTest

### PR DESCRIPTION
**Pull request by anmolnar**
Recommendation for refactoring `QuorumPeerMainTest`.
Based on the discussion from 
https://github.com/apache/zookeeper/pull/622#discussion_r218880876

### CODE REVIEW

**Review by eolivelli — APPROVED** [review]
Looka good to me, we can then strip out from this test the other methods, like the one I have added for metrics

File: src/java/test/org/apache/zookeeper/server/quorum/CloseSessionTxnTest.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import org.apache.zookeeper.ZooDefs;
+import org.apache.zookeeper.ZooKeeper;
+import org.apache.zookeeper.server.ZooKeeperServer;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+public class CloseSessionTxnTest extends QuorumPeerTestBase {
```
Comment (lvfangmin) [inline src/java/test/org/apache/zookeeper/server/quorum/CloseSessionTxnTest.java:32]: I can make this change and add the parameterized options in #622 after the QuorumPeerTestBase refactor part is landed here.

**Review by lvfangmin — APPROVED** [review]
This totally makes sense to me, actually I've done something similar in another patch I'm preparing by moving these common functions into the Base. In which I also enabled it to be able to launch observers in the LaunchServers function, I'll rebase that onto this when you merged it.

**Comment by anmolnar** [comment]
@lvfangmin @eolivelli What do you think guys? Shall we incorporate this into #622? 

Hearing that you've already prepared a separate patch just for the refactoring, I lean towards doing the refactor alone. In which case we need a separate Jira and the best would be to commit it as fast as possible after #622. :)

Makes sense?

**Comment by eolivelli** [comment]
I would prefer to do the refactor in a separate change.

Another thought is that if we always post pone clean ups we never do them :)

Anyway I see that we are in the middle of a big work of backporting changes, so I don''t know how much the refactor will be annoying for backports.

So my final opinion is: if this refactor does not make back port of FB changes more difficult lets do it as soon as possible

**Comment by lvfangmin** [comment]
@anmolnar our refactor is pretty similar, I'm fine to have this committed first just for the refactor part of QuorumPeerMainTest, I'll rebase #622 onto this after it's merged.

**Comment by anmolnar** [comment]
@lvfangmin Cool. Go ahead with the refactor first.

**Comment by lvfangmin** [comment]
@anmolnar did you have time to make the refactor? I'll rebase https://github.com/apache/zookeeper/pull/622 onto that when it's done.

**Comment by anmolnar** [comment]
@lvfangmin :)
Funny that seems like we're waiting for each other. I thought you were working on the refactor. :)
Let me create a separate Jira and submit a pull request quickly to get this done. Sorry!

**Pull request closed without merging**